CHANGES/4396.feature

```diff
@@ -0,0 +1 @@
+Add one-shot upload and accompanying tests and docs
```

Comment:
No need to mention tests in the changelog. IMO this should just be stuff we want users to know.